From: Felix Fietkau <nbd@nbd.name>
Date: Wed, 23 Oct 2024 18:50:10 +0200
Subject: [PATCH] nl80211: add new attributes for multi-radio support

- vif radio mask: used to assign vifs to specific radios
- monitor skip_tx flag: do not pass locally transmitted packets on the monitor interface
- radio antenna mask: radio specific part of the phy antenna mask

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---

--- a/lib/nl80211.c
+++ b/lib/nl80211.c
@@ -863,7 +863,7 @@ static const uc_nl_nested_spec_t nl80211
 
 static const uc_nl_nested_spec_t nl80211_msg = {
 	.headsize = 0,
-	.nattrs = 129,
+	.nattrs = 130,
 	.attrs = {
 		{ NL80211_ATTR_4ADDR, "4addr", DT_U8, 0, NULL },
 		{ NL80211_ATTR_AIRTIME_WEIGHT, "airtime_weight", DT_U16, 0, NULL },
@@ -994,6 +994,7 @@ static const uc_nl_nested_spec_t nl80211
 		{ NL80211_ATTR_MAX_AP_ASSOC_STA, "max_ap_assoc", DT_U16, 0, NULL },
 		{ NL80211_ATTR_SURVEY_INFO, "survey_info", DT_NESTED, 0, &nl80211_survey_info_nla },
 		{ NL80211_ATTR_WIPHY_RADIOS, "radios", DT_NESTED, DF_MULTIPLE|DF_AUTOIDX, &nl80211_wiphy_radio_nla },
+		{ NL80211_ATTR_VIF_RADIO_MASK, "vif_radio_mask", DT_U32, 0, NULL },
 	}
 };
 
